types/time.go

```diff
@@ -186,13 +186,14 @@ func FromGoTime(t gotime.Time) CoreTime {
 // FromDate makes a internal time representation from the given date.
 func FromDate(year int, month int, day int, hour int, minute int, second int, microsecond int) CoreTime {
 	t := ZeroCoreTime
-	t.setYear(uint16(year))
-	t.setMonth(uint8(month))
-	t.setDay(uint8(day))
-	t.setHour(uint8(hour))
-	t.setMinute(uint8(minute))
-	t.setSecond(uint8(second))
-	t.setMicrosecond(uint32(microsecond))
+	p := (*uint64)(&t)
```

Comment:
Please add comments for what will cause inlining not effective so that you used this way